keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+     * This operation can only be enabled on a soft-delete enabled vault. This operation requires the secrets/purge permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the RestResponse.
+     */
+    public Mono<RestResponse> purgeDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.purgeDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Recovers the deleted secret to the latest version.
+     * Recovers the deleted secret in the key vault. This operation can only be performed on a soft-delete enabled vault.
+     * This operation requires the secrets/recover permission.
+     *
+     * @param name The name of the deleted secret to be recovered.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the {@link Secret} that was recovered
+     */
+    public Mono<RestResponse<Secret>> recoverDeletedSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.recoverDeletedSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Backs up the specified secret.
+     * Requests that a backup of the specified secret be downloaded to the client. All versions of the secret will be downloaded.
+     * This operation requires the secrets/backup permission.
+     *
+     * @param name The name of the secret.
+     * @throws NullPointerException if {@code name} is {@code null}.
+     * @return the {@link SecretBackup} backup of the secret.
+     */
+    public Mono<RestResponse<SecretBackup>> backupSecretAsync(String name) {
+        Objects.requireNonNull(name, "The Secret name cannot be null.");
+        return service.backupSecret(vaultEndPoint, name, API_VERSION, ACCEPT_LANGUAGE);
+    }
+
+
+    /**
+     * Restores a backed up secret to a vault.
+     * Restores a backed up secret, and all its versions, to a vault.
+     * This operation requires the secrets/restore permission.
+     *
+     * @param backup The backup blob associated with the secret.
+     * @throws NullPointerException if {@code SecretBackup} is {@code null}.
+     * @return the {@link Secret} that was restored.
+     */
+    public Mono<RestResponse<Secret>> restoreSecretAsync(SecretBackup backup) {
+        Objects.requireNonNull(backup, "The Secret backup parameter cannot be null.");
+        Objects.requireNonNull(backup.value(), "The backup value cannot be null.");
+        SecretRestoreRequestOptions parameters = new SecretRestoreRequestOptions()
+                                                .withSecretBackup(backup.value());
+        return service.restoreSecret(vaultEndPoint, API_VERSION, ACCEPT_LANGUAGE, parameters);
+    }
+
+
+    /**
+     * List {@link Secret} secrets in a specified key vault.
```

Comment:
I think the JavaDoc might read better as `List {@link Secret secrets} in a specified key vault` (i.e. move the 'secrets' inside the braces, so that it replaces the Secret text and makes 'secret' the link text). However, on top of that, I don't like how it says "in a specified key vault" - because the method does not take a key vault argument. Perhaps the right text is "in the Key Vault instance represented by this client" or somesuch.

Reply:
updated, 
true replaced with "in the key vault"